sklearn/cluster/dbscan_.py

```diff
@@ -119,41 +129,14 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski',
                                 for neighbors in neighborhoods])
 
     # Initially, all samples are noise.
-    labels = -np.ones(X.shape[0], dtype=np.int)
+    labels = -np.ones(X.shape[0], dtype=np.intp)
 
     # A list of all core samples found.
-    core_samples = np.flatnonzero(n_neighbors > min_samples)
-    index_order = core_samples[random_state.permutation(core_samples.shape[0])]
-
-    # label_num is the label given to the new cluster
-    label_num = 0
-
-    # Look at all samples and determine if they are core.
-    # If they are then build a new cluster from them.
-    for index in index_order:
-        # Already classified
-        if labels[index] != -1:
-            continue
-
-        labels[index] = label_num
-
-        # candidates for new core samples in the cluster.
-        candidates = [index]
-        while len(candidates) > 0:
-            # The tolist() is needed for NumPy 1.6.
-            cand_neighbors = np.concatenate(np.take(neighborhoods, candidates,
-                                                    axis=0).tolist())
-            cand_neighbors = np.unique(cand_neighbors)
-            noise = cand_neighbors[labels.take(cand_neighbors) == -1]
-            labels[noise] = label_num
-            # A candidate is a core point in the current cluster that has
-            # not yet been used to expand the current cluster.
-            candidates = np.intersect1d(noise, core_samples,
-                                        assume_unique=True)
-        # Current cluster finished.
-        # Next core point found will start a new cluster.
-        label_num += 1
-    return core_samples, labels
+    core_samples = np.asarray(n_neighbors > min_samples, dtype=np.uint8)
```

Comment:
Both solutions allocate the temp array `n_neighbors > min_samples`. I don't see any advantage of one vs the other.